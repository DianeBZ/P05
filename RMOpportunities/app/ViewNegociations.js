import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  BackAndroid,
  ListView,
} from 'react-native';

import Translation from './Translation';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

//We change this and add some foncs to realise
//the display of the list of factures when we need.
var dataSourceVp = null;
var dataSourceAp = null;

export default class ViewNegociations extends Component {
  render() {
	BackAndroid.addEventListener('Back',this.onBackAndroid);
    return (
      <View style={styles.container}>
        <View style={{height:height*0.08}}/>
		<View style={styles.containerMain}>
			<View style = {styles.containerText}>
				<Text style = {styles.title}>
					{Translation[lang].nego}
				</Text>
				<Text style = {styles.instructions}>
					{Translation[lang].nego_text1}
				</Text>
			</View>

			<View style = {{justifyContent: 'flex-start',alignSelf: 'flex-start'}}>
				<Text style = {styles.titleAV}>
						{Translation[lang].vente_cours}
				</Text>
			</View>

			{(() => {
				if (dataSourceVp != null)
					return (
						<ListView />
					);
				else
					return (
						<View style = {styles.containerVentes}>
							<Text>
								{Translation[lang].aucune_vente}
							</Text>
						</View>
					);
			})()}

			<View style = {{justifyContent: 'flex-start',alignSelf: 'flex-start'}}>
				<Text style = {styles.titleAV}>
						{Translation[lang].achat_cours}
				</Text>
			</View>

			{(() => {
				if (dataSourceAp != null)
					return (
						<ListView />
					);
				else
					return (
						<View style = {styles.containerAchats}>
							<Text>
								{Translation[lang].aucun_achat}
							</Text>
						</View>
					);
			})()}
		</View>
      </View>
    );
  }
      onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  containerMain: {
    backgroundColor: '#F5FCFF',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.89,
	alignSelf: 'center',
  },
  containerText: {
    backgroundColor: '#F5FCFF',
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	flexDirection: 'column',
	width: width,
	height: height * 0.30,
	alignSelf: 'center',
  },
  containerVentes: {
	width: width,
	height: height * 0.23,
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	alignSelf: 'center',
	backgroundColor: '#F5FCFF',
	padding: 5,
	borderWidth: 1,
    borderRadius: 5,
	margin: 5,
    borderColor: '#CCC',
	width: 0.92 * width + 20,
  },
  containerAchats: {
	width: width,
	height: height * 0.23,
	flexGrow: 0,
	flexShrink: 0,
	justifyContent: 'center',
	alignItems: 'center',
	alignSelf: 'center',
	backgroundColor: '#F5FCFF',
	padding: 5,
	borderWidth: 1,
    borderRadius: 5,
	margin: 5,
    borderColor: '#CCC',
	width: 0.92 * width + 20,
  },
  title: {
    fontSize: 40,
    textAlign: 'center',
    margin: 10,
	color: '#333333',
  },
  titleAV: {
	fontSize: 20,
	color: '#333333',
	textAlign: 'left',
  },
  instructions: {
    fontSize: 15,
	textAlign: 'center',
    marginBottom: 5,
  },
});