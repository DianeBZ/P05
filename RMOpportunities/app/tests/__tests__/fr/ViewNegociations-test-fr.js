// __tests__/ViewNegociations-test-fr.js
import 'react-native';
import React from 'react';

global.lang = 'fr';
import ViewNegociations from '../../../main/ViewNegociations';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <ViewNegociations />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
