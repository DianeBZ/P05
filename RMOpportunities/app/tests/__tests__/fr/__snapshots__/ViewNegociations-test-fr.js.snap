// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#F2F2F2",
        "flexDirection": "column",
        "flexGrow": 0,
        "flexShrink": 0,
        "height": 1187.26,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "flexDirection": "column",
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 400.2,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 40,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Mes négociations
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 15,
            "marginBottom": 5,
            "textAlign": "center",
          }
        }
      >
        Visionner vos négociations en cours
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 20,
            "textAlign": "left",
          }
        }
      >
        Ventes en cours
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 306.82,
          "justifyContent": "center",
          "margin": 5,
          "padding": 5,
          "width": 710,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Aucune vente
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 20,
            "textAlign": "left",
          }
        }
      >
        Achats en cours
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#F2F2F2",
          "borderColor": "#CCC",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexGrow": 0,
          "flexShrink": 0,
          "height": 306.82,
          "justifyContent": "center",
          "margin": 5,
          "padding": 5,
          "width": 710,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Aucun achat
      </Text>
    </View>
  </View>
</View>
`;
